openstudiocore/src/osversion/VersionTranslator.cpp

```diff
@@ -4490,6 +4491,72 @@ std::string VersionTranslator::update_2_7_1_to_2_7_2(const IdfFile& idf_2_7_1, c
 
 }
 
+std::string VersionTranslator::update_2_8_0_to_2_8_1(const IdfFile& idf_2_8_0, const IddFileAndFactoryWrapper& idd_2_8_1) {
+  std::stringstream ss;
+  
+  ss << idf_2_8_0.header() << std::endl << std::endl;
+  IdfFile targetIdf(idd_2_8_1.iddFile());
+  ss << targetIdf.versionObject().get();
+  
+  boost::optional<std::string> value;
+  boost::optional<double> fraction;
+  
+  for (const IdfObject& object : idf_2_8_0.objects()) {
+    auto iddname = object.iddObject().name();
+
+    if (iddname == "OS:ZoneHVAC:EquipmentList") {
+      auto iddObject = idd_2_8_1.getObject("OS:ZoneHVAC:EquipmentList");
+      IdfObject newObject(iddObject.get());
+
+      // Copy non extensible fields in place
+      for (size_t i = 0; i < object.numNonextensibleFields(); ++i) {
+        if( value = object.getString(i) ) {
+          newObject.setString(i, value.get());
+        }
+      }
+
+      // Copy the eg values 0 through 2 and create ScheduleConstant out of 3 and 4
+      for (const IdfExtensibleGroup& eg : object.extensibleGroups()) {
+        IdfExtensibleGroup new_eg = newObject.pushExtensibleGroup();
+        for (size_t i = 0; i < 5; ++i) {
+          if( i > 2 ) {            
+            if( fraction = eg.getDouble(i) ) {
+              auto iddObject = idd_2_8_1.getObject("OS:Schedule:Constant");
+              IdfObject scheduleConstant(iddObject.get());
+              std::string uuid = toString(createUUID());
+              scheduleConstant.setString(0, uuid);
+              scheduleConstant.setString(1, "Name");
+              for (const IdfObject& object2 : idf_2_8_0.objects()) {
+                if ( object2.getString(0).get() == eg.getString(0).get() ) {
```

Comment:
Generally speaking, use `openstudio::istringEqual` to do a case insensitive comparison instead please, like E+ does.

Also, please do **not** loop on all objects here. You can get the object by it's handle directly via `boost::optional<IdfObject> IdfFile:getObject(const Handle& handle) const;`